platforms/documentation/docs/src/docs/userguide/best-practices/bp_general.adoc

```diff
@@ -57,3 +57,39 @@ include::sample[dir="snippets/bestPractices/usePluginsBlock/groovy/do-this",file
 
 === Tags
 `#structuring-builds`
+
+[[avoid_duplicate_dependencies]]
+== Avoid Declaring Dependencies Multiple Times
+
+You should avoid duplicating dependencies in your build scripts, whether with different versions or in different dependency configurations.
```

Comment:
I wonder if this is good _Gradle_ advice rather that good _general software development _ advice.

I think something like "don't duplicate already transitively present deps with direct declarations" would be more Gradle-specific, as such advice wouldn't necessarily apply to every build tool.

Reply:
Also something like don't re-declare deps already present in extended configurations, like re-declaring something in `compileOnly` that's already in `implementation`.

I think you could probably cover all of these concerns in this single item.